app/slashcommands/command_custom_status.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package slashcommands
+
+import (
+	"strings"
+
+	goi18n "github.com/mattermost/go-i18n/i18n"
+
+	"github.com/mattermost/mattermost-server/v5/app"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type CustomStatusProvider struct {
+}
+
+const (
+	CMDCustomStatus      = "status"
+	CMDCustomStatusClear = "clear"
+
+	DefaultCustomStatusEmoji = "speech_balloon"
+)
+
+func init() {
+	app.RegisterCommandProvider(&CustomStatusProvider{})
+}
+
+func (*CustomStatusProvider) GetTrigger() string {
+	return CMDCustomStatus
+}
+
+func (*CustomStatusProvider) GetCommand(a *app.App, T goi18n.TranslateFunc) *model.Command {
+	return &model.Command{
+		Trigger:          CMDCustomStatus,
+		AutoComplete:     true,
+		AutoCompleteDesc: T("api.command_custom_status.desc"),
+		AutoCompleteHint: T("api.command_custom_status.hint"),
+		DisplayName:      T("api.command_custom_status.name"),
+	}
+}
+
+func (*CustomStatusProvider) DoCommand(a *app.App, args *model.CommandArgs, message string) *model.CommandResponse {
+	if message == CMDCustomStatusClear {
+		if err := a.RemoveCustomStatus(args.UserId); err != nil {
+			mlog.Error(err.Error())
+			return &model.CommandResponse{Text: args.T("api.command_custom_status.clear.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+		}
+
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         args.T("api.command_custom_status.clear.success"),
+		}
+	}
+
+	customStatus := &model.CustomStatus{
+		Emoji: DefaultCustomStatusEmoji,
+		Text:  message,
+	}
+	firstEmojiLocations := model.EMOJI_PATTERN.FindIndex([]byte(message))
```

Comment:
@reflog @jespino Maybe you can pitch in here.

There is an issue with t

Reply:
@chetanyakan I would probably fix the model.EMOJI_PATTERN itself, if it's actually an incorrect regex. I doubt that will break things.
@jespino - thoughts?

Reply:
The pattern is technically correct since it was written to be used for custom emojis, and like you found, `:+1:` is a bit of a special case there. That said, it keeps causing us problems like this, so I think we allow plus signs in custom emojis to simplify things. I have wanted to make system emojis more similar to custom emojis for a while now. I think we'd need to consider what else needs to change for this though since at least the web app and some help text would need updates from what I see.

Maybe for now, we could add a new pattern for use here? It's not great, and I imagine naming it something like `ALL_EMOJI_PATTERN` might confuse some people in the future, but I think that's the safest thing to do for now.